File: agent-collab-system/backend/python-agent-service/app/mock_records/schemas/records.py
from pydantic import BaseModel, Field


class MockRecordsSourceItem(BaseModel):
    source_system: str
    label: str


class MockRecordsTableSchemaField(BaseModel):
    name: str
    label: str
    field_type: str
    editable: bool = True


class MockRecordsTableSchemaResponse(BaseModel):
    table_name: str
    label: str
    source_system: str
    fields: list[MockRecordsTableSchemaField] = Field(default_factory=list)


class MockRecordsTableItem(BaseModel):
    table_name: str
    label: str
    source_system: str
    description: str


class MockRecordsSourcesResponse(BaseModel):
    sources: list[MockRecordsSourceItem] = Field(default_factory=list)


class MockRecordsTablesResponse(BaseModel):
    tables: list[MockRecordsTableItem] = Field(default_factory=list)


class MockRecordsRowListResponse(BaseModel):
    table_name: str
    rows: list[dict[str, object]] = Field(default_factory=list)


class MockRecordUpsertRequest(BaseModel):
    values: dict[str, object] = Field(default_factory=dict)


class MockRecordMutationResponse(BaseModel):
    table_name: str
    record_id: str
    row: dict[str, object]
    change_event_id: str
    triggered_execution_ids: list[str] = Field(default_factory=list)


class MockRecordsRecentEventItem(BaseModel):
    change_event_id: str
    table_name: str
    record_id: str
    operation: str
    event_type: str
    changed_fields: list[str] = Field(default_factory=list)
    triggered_execution_ids: list[str] = Field(default_factory=list)
    created_at: str | None = None


class MockRecordsRecentEventsResponse(BaseModel):
    events: list[MockRecordsRecentEventItem] = Field(default_factory=list)

File: agent-collab-system/frontend/src/types/records.ts
export interface RecordsSourceItem {
  source_system: string
  label: string
}

export interface RecordsSourcesResponse {
  sources: RecordsSourceItem[]
}

export interface RecordsTableItem {
  table_name: string
  label: string
  source_system: string
  description: string
}

export interface RecordsTablesResponse {
  tables: RecordsTableItem[]
}

export interface RecordsTableSchemaField {
  name: string
  label: string
  field_type: string
  editable: boolean
}

export interface RecordsTableSchemaResponse {
  table_name: string
  label: string
  source_system: string
  fields: RecordsTableSchemaField[]
}

export interface RecordsRowListResponse {
  table_name: string
  rows: Record<string, unknown>[]
}

export interface RecordMutationResponse {
  table_name: string
  record_id: string
  row: Record<string, unknown>
  change_event_id: string
  triggered_execution_ids: string[]
}

export interface RecentRecordEvent {
  change_event_id: string
  table_name: string
  record_id: string
  operation: string
  event_type: string
  changed_fields: string[]
  triggered_execution_ids: string[]
  created_at?: string | null
}

export interface RecentRecordEventsResponse {
  events: RecentRecordEvent[]
}
